blockmanager.go

```diff
@@ -1041,9 +1075,8 @@ func (b *blockManager) handleInvMsg(imsg *invMsg) {
 		case wire.InvTypeBlock:
 			// Request the block if there is not already a pending
 			// request.
-			if _, exists := b.requestedBlocks[iv.Hash]; !exists {
-				b.requestedBlocks[iv.Hash] = struct{}{}
-				imsg.peer.requestedBlocks[iv.Hash] = struct{}{}
+			if _, exists := b.peers[imsg.peer.ID()].requestedBlocks[iv.Hash]; !exists {
```

Comment:
Rather than looking up `b.peers[imsg.peer.ID()]` multiple times throughout the function, why not just get a pointer to the peer data once and use the local throughout the function?

```Go
peerState := b.peers[imsg.peer.ID()]
...
case wire.InvTypeBlock:
	_, exists := peerState.requestedblocks[iv.Hash]
	if !exists {
		...
```